experiments/GCM/heldsuarez.jl

```diff
@@ -0,0 +1,140 @@
+    # callable to set initial conditions
+    FT = eltype(state)
+
+    r = norm(coords, 2)
+    h = r - FT(planet_radius)
+
+    scale_height = R_d * setup.T_initial / grav
+    p            = setup.p_ground * exp(-h / scale_height)
+
+    state.ρ  = air_density(setup.T_initial, p)
+    state.ρu = SVector{3, FT}(0, 0, 0)
+    state.ρe = state.ρ * (internal_energy(setup.T_initial) + aux.orientation.Φ)
+
+    return nothing
+end
+
+function config_heldsuarez(FT, N, resolution, setup)
+    config = CLIMA.GCM_Configuration("HeldSuarez", FT, ArrayType, setup,
+                                     ref_state=HydrostaticState(IsothermalProfile(setup.T_initial), FT(0)),
+                                     turbulence=ConstantViscosityWithDivergence(FT(0)),
+                                     moisture=DryModel(),
+                                     sources=(Gravity(),
+                                              Coriolis(),
+                                              held_suarez_forcing!))
+    CLIMA.GCM_SetupGrid!(config, N, resolution, setup.domain_height)
+
+    return config
+end
+
+function held_suarez_forcing!(source, state, aux, t::Real)
+    FT = eltype(state)
+
+    ρ     = state.ρ
+    ρu    = state.ρu
+    ρe    = state.ρe
+    coord = aux.coord
+    Φ     = aux.orientation.Φ
+    e     = ρe / ρ
+    u     = ρu / ρ
+    e_int = e - u' * u / 2 - Φ
+    T     = air_temperature(e_int)
+
+    # Held-Suarez constants
+    k_a       = FT(1 / (40 * day))
+    k_f       = FT(1 / day)
+    k_s       = FT(1 / (4 * day))
+    ΔT_y      = FT(60)
+    Δθ_z      = FT(10)
+    T_equator = FT(315)
+    T_min     = FT(200)
+
+    σ_b          = FT(7 / 10)
+    r            = norm(coord, 2)
+    @inbounds λ  = atan(coord[2], coord[1])
+    @inbounds φ  = asin(coord[3] / r)
+    h            = r - FT(planet_radius)
+    scale_height = FT(7000) # from Smolarkiewicz JAS 2001 paper
+    σ            = exp(-h / scale_height)
+
+    # TODO: use
+    #  p = air_pressure(T, ρ)
+    #  σ = p/p0
+    exner_p       = σ ^ (R_d / cp_d)
+    Δσ            = (σ - σ_b) / (1 - σ_b)
+    height_factor = max(0, Δσ)
+    T_equil       = (T_equator - ΔT_y * sin(φ) ^ 2 - Δθ_z * log(σ) * cos(φ) ^ 2 ) * exner_p
+    T_equil       = max(T_min, T_equil)
+    k_T           = k_a + (k_s - k_a) * height_factor * cos(φ) ^ 4
+    k_v           = k_f * height_factor
+
+    source.ρu += -k_v * ρu
+    source.ρe += -k_T * ρ * cv_d * (T - T_equil) - k_v * ρu' * ρu / ρ
+end
+
+function main()
+    CLIMA.init()
+
+    FT = Float64
+
+    # DG polynomial order
+    N = 5
+
+    # Domain resolution
+    nelem_horz = 6
+    nelem_vert = 8
+    resolution = (nelem_horz, nelem_vert)
+
+    setup = HeldSuarezSetup{FT}()
+
+    # Set up the driver configuration
+    config = config_heldsuarez(FT, N, resolution, setup)
+
+    # Determine the time step
+    element_size = (setup.domain_height / nelem_vert)
+    acoustic_speed = soundspeed_air(FT(315))
+    lucas_magic_factor = 5
```

Comment:
```suggestion
    Courant_number = 0.8
```
(I'm not sure what the right Courant number here is, but it should be less than one.)

Reply:
Why should it be less than one?

Reply:
From what @smarras79 reports, ARK232 gave best overall performance. @smarras79: what was the horizontal Courant number you used? More generally, we need (a) guidance about stability regions of the various schemes in the design docs, and (b) should associate default Courant numbers to the different schemes, so that the default Courant number decision is not made on the user (driver) level.

Reply:
As of now, the maximum stable Courant number for my runs was 0.2 or less using **LSRK54CarpenterKennedy** but have not yet looked at the max stable values for **ARK2GiraldoKellyConstantinescu** 

I will get back to you on this for values.
This said, I have been in touch with @fxgiraldo about 1D IMEX for dycoms because the solution deviates from the correct one. I am waiting for his corrections on this.

Reply:
@smarras79 @lcw @fxgiraldo Could you guys please help Kiran figure out how to set up the various time steppers so that they are associated with default Courant numbers based on their stability regions, and how to incorporate this into the configuration files? We need to make these configuration decisions less "magic" and more transparent. The configurations should also be flexible enough that we can (soon, I hope) accommodate IMEX with multirate, without the user having to specify details at the driver level.

Reply:
I think https://github.com/climate-machine/CLIMA/pull/593 is introducing this. I'll work with @smarras79 to fold this in when it is worked out.

Reply:
Hi @kpamnany, the Courant calculator is in place in the PR that you indicated, but I am working with @fxgiraldo to get the 1D IMEX best settings for our use. 
@fxgiraldo please, let me know how I can help you to address this.